Automattic/jetpack
# WAF: bugfix header handling, remove old file

#### Changes proposed in this Pull Request:

* When the package was created, `normalizeHeaderName()` was renamed to `normalize_header_name()` but one instance was missed; this leads to fatals with rules that check header values. This PR completes that rename.
* Additionally, the `substr()` that strips the `http-` from the header names was operating in reverse, causing all of them to be stored under `http-` in the resulting array and overwriting each other. This PR adjusts that call.
* Finally, there was an unused file from our prototype work that was included in the package; this removes it.

#### Jetpack product discussion

1201069996155224-as-1201880901325575




#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Without this branch applied, add the following to the end of the `rules.php` file in this package:
```
$rule = (object) array( 'id' => 133700010, 'reason' => '', 'tags' => array (
        0 => 'stuff',
) );
if($waf->match_targets(array (
),array (
  'request_headers' => 
  array (
    'count' => true,
  ),
),'eq','10',false,true)) {
        $rule->reason = 'Intentionally blocking this test-request. query_string, eq';
        return $waf->block('block',$rule->id,$rule->reason,403);
}
```
* Rebuild as needed, and go to any URL on your testing site; it should error, and you should be able to see messages in the PHP error logs confirming the issue is `normalizeHeaderNames()`.
* Apply this PR, rebuild as needed, and test again; there should be no more fatal errors.
* To test the `substr()` changes, you can add the line `error_log( var_export( $value, true ) );` before the `break;` line at 384; hit the URL again and it should log a full set of headers (with this PR) or only a few (without this PR).
